android/src/main/java/com/polidea/flutter_ble_lib/FlutterBleLibPlugin.java

```diff
@@ -61,7 +61,7 @@ public static void registerWith(Registrar registrar) {
         final EventChannel connectionStateChannel = new EventChannel(registrar.messenger(), ChannelName.CONNECTION_STATE_CHANGE_EVENTS);
         final EventChannel characteristicMonitorChannel = new EventChannel(registrar.messenger(), ChannelName.MONITOR_CHARACTERISTIC);
 
-        final FlutterBleLibPlugin plugin = new FlutterBleLibPlugin(registrar.activity().getApplicationContext());
+        final FlutterBleLibPlugin plugin = new FlutterBleLibPlugin(registrar.context());
```

Comment:
Why not `getApplicationContext()`?

Reply:
This is, in fact, application context.
https://api.flutter.dev/javadoc/io/flutter/plugin/common/PluginRegistry.Registrar.html#context--

Reply:
We are not yet up to the new v2 API I suppose?

Reply:
Maybe it's a good idea to migrate android plugin to v2 as it's a minor release? What do you guys think?

Reply:
We can do that in `2.2.0` no worries ;)